src/operators.jl

```diff
@@ -107,3 +107,56 @@ function Op(f, T=Any; iscommutative=false)
     finalizer(free, op)
     return op
 end
+
+"""
+    @Op(f, T)
```

Comment:
It'd be nice to add a reference to `@Op` in `Op`'s docstring

Reply:
Maybe I should also call it "RegisterOp" or something like that?

Reply:
Uhm, maybe yes, it'd hint better they aren't exactly working at the same level